AKS: support for ingress_application_gateway addon (#11376)
terraform-providers/terraform-provider-azurerm: website/docs/r/kubernetes_cluster.html.markdown

---

An `ingress_application_gateway` block supports the following:

> * `enabled` - (Required) Set to true to deploy the application gateway ingress controller to this cluster.

Reviewer: Can we modify the wording a bit here?

```suggestion
* `enabled` - (Required) Whether to deploy the Application Gateway ingress controller to this Kubernetes Cluster?
```

> * `application_gateway_id` - (Optional) Integrate the application gateway ingress controller with the application gateway with the given ID. See [this](https://docs.microsoft.com/en-us/azure/application-gateway/tutorial-ingress-controller-add-on-existing) page for further details.

Reviewer: Can we modify the wording a bit here?

```suggestion
* `application_gateway_id` - (Optional) The ID of the Application Gateway that to integrate with the ingress controller of this Kubernetes Cluster. See [this](https://docs.microsoft.com/en-us/azure/application-gateway/tutorial-ingress-controller-add-on-existing) page for further details.
```

> * `subnet_cidr` - (Optional) Create an application gateway using the subnet CIDR and then integrate the application gateway ingress controller with the gateway. See [this](https://docs.microsoft.com/en-us/azure/application-gateway/tutorial-ingress-controller-add-on-new) page for further details.

Reviewer: ```suggestion
* `subnet_cidr` - (Optional) The subnet CIDR that be used to create an Application Gateway, which in turns will be integrated with the ingress controller of this Kubernetes Cluster. See [this](https://docs.microsoft.com/en-us/azure/application-gateway/tutorial-ingress-controller-add-on-new) page for further details.
```

> * `subnet_id` - (Optional) Create an application gateway on the subnet with the given ID and then integrate the application gateway ingress controller with the gateway. See [this](https://docs.microsoft.com/en-us/azure/application-gateway/tutorial-ingress-controller-add-on-new) page for further details.

Reviewer: ```suggestion
* `subnet_id` - (Optional) The ID of the Subnet to create an Application Gateway, which in turns will be integrated with the ingress controller of this Kubernetes Cluster. See [this](https://docs.microsoft.com/en-us/azure/application-gateway/tutorial-ingress-controller-add-on-new) page for further details.
```

}
```

---

The `addon_profile` block exports the following:

* `ingress_application_gateway` - An `ingress_application_gateway` block as defined below.

---

The `ingress_application_gateway` block exports the following:

> * `effective_application_gateway_id` - The ID of the application gateway associated with the application gateway ingress controller deployed to this cluster.

Reviewer: ```suggestion
* `effective_application_gateway_id` - The ID of the Application Gateway associated with the ingress controller deployed to this Kubernetes Cluster.
```

}
```

---

The `addon_profile` block exports the following:

* `ingress_application_gateway` - An `ingress_application_gateway` block as defined below.

---

The `ingress_application_gateway` block exports the following:

* `effective_gateway_id` - The ID of the Application Gateway associated with the ingress controller deployed to this Kubernetes Cluster.

* `gateway_id` - The ID of the Application Gateway integrated with the ingress controller of this Kubernetes Cluster. This attribute is only set when gateway_id is specified when configuring the `ingress_application_gateway` addon.

* `subnet_cidr` - The subnet CIDR used to create an Application Gateway, which in turn will be integrated with the ingress controller of this Kubernetes Cluster. This attribute is only set when `subnet_cidr` is specified when configuring the `ingress_application_gateway` addon.

> * `subnet_id` - The ID of the subnet on which to create an Application Gateway, which in turn will be integrated with the ingress controller of this Kubernetes Cluster. This attribute is only set when `subnet_id` is specified when configuring the `ingress_application_gateway` addon.

Reviewer: Since these are input only properties, shall we remove them here as this section shall only contain the `Computed` properties?
